src/api/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { ApiState } from '../reducers/api';
+
+export enum HttpMethod {
+  DELETE = 'DELETE',
+  GET = 'GET',
+  PATCH = 'PATCH',
+  POST = 'POST',
+  PUT = 'PUT',
+}
+
+type CallApiParams = {
+  endpoint: string;
+  method?: HttpMethod;
+  credentials?: boolean;
+  version?: string;
+  apiState: ApiState;
+};
+
+type Headers = {
+  [name: string]: string;
+};
+
+export const callApi = async ({
+  apiState,
+  endpoint,
+  credentials = false,
+  method = HttpMethod.GET,
+  version = 'v4',
+}: CallApiParams) => {
+  if (!endpoint.startsWith('/')) {
+    endpoint = `/${endpoint}`;
+  }
+  if (!endpoint.endsWith('/')) {
+    endpoint = `${endpoint}/`;
+  }
+
+  const headers = {} as Headers;
+  if (apiState.authToken) {
+    headers['Authorization'] = `Bearer ${apiState.authToken}`;
+  }
+
+  try {
+    const response = await fetch(`/api/${version}${endpoint}`, {
+      method,
+      headers,
+      credentials: credentials ? 'include' : undefined,
```

Comment:
I'm confused about credentials (I always have been on `addons-frontend`). I think we should not let it be configurable and always set it to some value -- either `include` or `undefined` -- until we run into problems. By not passing credentials, the API won't be able to recognize its own cookies. Does it need to see its own cookies? I'm not sure. If I toggle `credentials` on and off, the current log out call you added here still works. I guess that's because no cookies are involved.

Reply:
In general, we should not send the cookies, especially because the API does not read them.

>  If I toggle `credentials` on and off, the current log out call you added here still works. I guess that's because no cookies are involved.

Ah? Really? The API returns a `set-cookie` header (to remove the cookies) and, without `credentials: include`, this header does not seem to be taken into account by the browser: no cookie gets removed without `credentials: include`.